frappe/public/js/frappe/form/sidebar/share.js

```diff
@@ -187,4 +191,52 @@ frappe.ui.form.Share = class Share {
 			});
 		});
 	}
+	async share_modal() {
+		const default_expiry = frappe.boot.sysdefaults.document_share_key_expiry;
+		const share_modal = new frappe.ui.Dialog({
+			title: __("Share Link"),
+			fields: [{
+				fieldname: "link_expiration_date",
+				label: __("Link Expiration Date"),
+				fieldtype: "Date",
+				default: frappe.datetime.add_days(moment(), default_expiry),
+				min_date: new Date(),
+				description: __("The default expiration date is {0} days from today.", [default_expiry]),
+				read_only_depends_on: "eval: doc.link"
+			}, {
+				fieldtype: "Check",
+				label: __("No Expiry"),
+				fieldname: "no_expiry",
+				change: () => {
+					const no_expiry_warning = `<span style="color: var(--red-500);">
+						${__('It is not recommended to send links without expiry')}
```

Comment:
Is this needed? Maybe make it a warning. Perhaps it would be better to say why or just warn that the link user will be able to see all changes to this document as long as it is not deleted.